tests/test_client_functional.py

```diff
@@ -3688,6 +3689,42 @@ async def handler(request: web.Request) -> web.Response:
         await resp.read()
 
 
+async def test_read_after_catch_raise_for_status(aiohttp_client: AiohttpClient) -> None:
+    async def handler(request: web.Request) -> web.Response:
+        return web.Response(body=b"data", status=404)
+
+    app = web.Application()
+    app.add_routes([web.get("/", handler)])
+
+    client = await aiohttp_client(app)
+
+    async with client.get("/") as resp:
+        with pytest.raises(ClientResponseError):
+            # Should not release response when in async with context.
+            await resp.raise_for_status()
+
+        result = await resp.read()
+        assert result == b"data"
+
+
+async def test_read_after_raise_outside_context(aiohttp_client: AiohttpClient) -> None:
+    async def handler(request: web.Request) -> web.Response:
+        return web.Response(body=b"data", status=404)
+
+    app = web.Application()
+    app.add_routes([web.get("/", handler)])
+
+    client = await aiohttp_client(app)
+
+    resp = await client.get("/")
+    with pytest.raises(ClientResponseError):
```

Comment:
Plz always use the `match=` regex with `pytest.raises()`, `pytest.warns()` and `pytest.deprecated()`. This improves the accuracy of such tests and reduces false-positives.